### ISSUE TITLE
[question] Difficulties using with typescript

### ISSUE THREAD

**Issue by DrakoPOD** [moscajs/aedes#683]
I have this code, that works perfectly in vanilla JS:
```js
aedes.on('publish', function (packet, client) {
	//More code...
	aedes.publish({
		topic: 'auth/' + client.id,
		payload: '1',
		qos: 0
	});
});
```
But with typescript it says that is missing one argument, in that case is `callback`, Then, I put the callback, but get another error that says (translate from Spanish):

> Cannot assign an arg of type "{ topic: string; payload: string; qos: 0; }" to param of type "PublishPacket".
> to type "{ topic: string; payload: string; qos: 0; }" is missing the following properties of type "IPublishPacket": cmd, dup, retain

It suppose that these properties should be defaults or optional, also the docs doesn't offer explanation of what properties have or what is `PublishPacket`. I know that properties are standard of mqtt, but at least a external link or something.

Also, I got another error with `aedes.authenticate`
`'authenticate' does not exist on type 'aedes'`

The code can run, at least I don't use typescript. Adding `// @ts-expect-error` ignore the problem. But, how I can import the type definitions?

**Comment by robertsLando** [comment]
@DrakoPOD you can add the authenticate function in aedes options

**Comment by robertsLando** [comment]
Fixed by #687
